util/metadata_service/partition.py

```diff
@@ -109,7 +141,40 @@ def get_first_before_metadata(stream_key, start_time):
     """
     res = _query_partition_metadata_before(stream_key, start_time)
     # filter to ensure start time < time_range_start
-    res = filter(lambda x: x.first <= start_time, res)
+    res = filter(lambda x: x.first < start_time, res)
+    if not res:
+        return {}
+    first_bin = res[0].bin
+    res = filter(lambda x: x.bin == first_bin, res)
+    if len(res) == 1:
+        to_use = res[0]
+    else:
+        # Check same size
+        if res[0].count == res[1].count:
+            # take the choosen one
+            res = filter(lambda x: x.store == engine.app.config['PREFERRED_DATA_LOCATION'], res)
+            to_use = res[0]
+        # other otherwise take the larger of the two
```

Comment:
remove extra "other" at start of comment
ditto for get_first_after_metadata

Reply:
Done